gridstatus/ercot.py

```diff
@@ -432,42 +509,257 @@ def get_interconnection_queue(self, verbose=False):
 
         return queue
 
+    def get_spp(
+        self,
+        date,
+        end=None,
+        market: str = None,
+        locations: list = None,
+        location_type: str = None,
+        verbose=False,
+    ):
+        """Get SPP data for ERCOT
+
+        Supported Markets: REAL_TIME_15_MIN, DAY_AHEAD_HOURLY
+
+        Supported Location Types: "zone", "hub", "node"
+        """
+        if locations is None:
+            locations = "ALL"
+
+        if location_type is None:
+            location_type = LOCATION_TYPE_ZONE
+
+        assert market is not None, "market must be specified"
+        market = Markets(market)
+
+        if market == Markets.REAL_TIME_15_MIN:
+            return self._get_spp_rtm15(
+                date,
+                locations,
+                location_type,
+                verbose,
+            )
+        elif market == Markets.DAY_AHEAD_HOURLY:
+            return self._get_spp_dam(date, locations, location_type, verbose)
```

Comment:
this is small, but I think you should do the call to `Ercot._finalize_spp_df` here since both dam and rtm call it. 

what I am thinking

```python
if market == Markets.REAL_TIME_15_MIN:
    df = self._get_spp_rtm15(
        date,
        locations,
        location_type,
        verbose,
    )
elif market == Markets.DAY_AHEAD_HOURLY:
    df = self._get_spp_dam(date, locations, location_type, verbose)

else:
    raise NotSupported(
        f"Market {market} not supported for ERCOT",
    )
    
return Ercot._finalize_spp_df(df, "SettlementPoint", locations)
```

Reply:
Sounds good

Reply:
Oh right, I couldn't do that because the settlement point fields are named differently, because of course:

```
ercot.py
585:        return Ercot._finalize_spp_df(df, "SettlementPoint", locations)
695:        return Ercot._finalize_spp_df(df, "SettlementPointName", locations)
```

I could do something like this:
```python
if market == Markets.REAL_TIME_15_MIN:
    df = self._get_spp_rtm15(
        date,
        locations,
        location_type,
        verbose,
    )
    settlement_point_field = "SettlementPoint"
elif market == Markets.DAY_AHEAD_HOURLY:
    df = self._get_spp_dam(date, locations, location_type, verbose)
    settlement_point_field = "SettlementPointName"
else:
    raise NotSupported(
        f"Market {market} not supported for ERCOT",
    )
    
return Ercot._finalize_spp_df(df, settlement_point_field, locations)
```

Does that work?

Reply:
ahh, i see. i prefer to have the `finalize_spp_df` call in this function since it makes it more clear its common to all the markets. i think your approach works.